digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/RoadWidthService.scala

```diff
@@ -25,47 +24,50 @@ class RoadWidthService(roadLinkServiceImpl: RoadLinkService, eventBusImpl: Digir
 
-    val linkIds = roadLinks.map(_.linkId)
-    val mappedChanges = LinearAssetUtils.getMappedChanges(changes)
-    val removedLinkIds = LinearAssetUtils.deletedRoadLinkIds(mappedChanges, linkIds.toSet)
-    val existingAssets =
-      withDynTransaction {
-          dao.fetchLinearAssetsByLinkIds(LinearAssetTypes.RoadWidthAssetTypeId, linkIds ++ removedLinkIds, LinearAssetTypes.numericValuePropertyId)
-      }
+      val linkIds = roadLinks.map(_.linkId)
+      val mappedChanges = LinearAssetUtils.getMappedChanges(changes)
+      val removedLinkIds = LinearAssetUtils.deletedRoadLinkIds(mappedChanges, linkIds.toSet)
+      val existingAssets =
+        withDynTransaction {
+            dao.fetchLinearAssetsByLinkIds(LinearAssetTypes.RoadWidthAssetTypeId, linkIds ++ removedLinkIds, LinearAssetTypes.numericValuePropertyId)
+        }
 
-    val timing = System.currentTimeMillis
-    val (assetsOnChangedLinks, assetsWithoutChangedLinks) = existingAssets.partition(a => LinearAssetUtils.newChangeInfoDetected(a, mappedChanges))
+      val timing = System.currentTimeMillis
+      val (assetsOnChangedLinks, assetsWithoutChangedLinks) = existingAssets.partition(a => LinearAssetUtils.newChangeInfoDetected(a, mappedChanges))
 
-    val projectableTargetRoadLinks = roadLinks.filter(rl => rl.linkType.value == UnknownLinkType.value || rl.isCarTrafficRoad)
+      val projectableTargetRoadLinks = roadLinks.filter(rl => rl.linkType.value == UnknownLinkType.value || rl.isCarTrafficRoad)
 
-    val (expiredIds, newRoadWidthAssets) = getRoadWidthAssetChanges(existingAssets, roadLinks, changes, (newAssetIds) => withDynTransaction {
-      dao.fetchExpireAssetLastModificationsByLinkIds(LinearAssetTypes.RoadWidthAssetTypeId, newAssetIds)
-    })
-    val initChangeSet = ChangeSet(droppedAssetIds = Set.empty[Long],
-                                  expiredAssetIds = (existingAssets.filter(asset => removedLinkIds.contains(asset.linkId)).map(_.id).toSet ++ expiredIds).filterNot(_ == 0L),
-                                  adjustedMValues = Seq.empty[MValueAdjustment],
-                                  adjustedSideCodes = Seq.empty[SideCodeAdjustment])
+      val initChangeSet: ChangeSet = ChangeSet(droppedAssetIds = Set.empty[Long],
+                                    expiredAssetIds = existingAssets.filter(asset => removedLinkIds.contains(asset.linkId)).map(_.id).toSet.filterNot(_ == 0L),
+                                    adjustedMValues = Seq.empty[MValueAdjustment],
+                                    adjustedSideCodes = Seq.empty[SideCodeAdjustment])
 
-    val combinedAssets = assetsOnChangedLinks.filterNot(a => expiredIds.contains(a.id)) ++ newRoadWidthAssets
+      val (projectedAssets, changedSetProjected) = fillNewRoadLinksWithPreviousAssetsData(projectableTargetRoadLinks,
+        assetsOnChangedLinks, assetsOnChangedLinks, changes, initChangeSet)
```

Comment:
We are sending the same Sequence (assetsOnChangedLinks) on assetsToUpdate and currentAssets parameters of fillNewRoadLinksWithPreviousAssetsData function.
Maybe we can send an empty sequence to one of them.

Reply:
In this case we didn't have any difference between 'assetsToUpdate' and 'currentAssets' . For that reason 'assetsOnChangedLinks' is sent twice.